### ISSUE TITLE
Generated codes for nullable Enum got analysis error

### ISSUE THREAD

**Issue by xiaozhizhong** [pinchbv/floor#721]
After updated to V1.4.0, I defined a nullable Enum type in Entity, but got analysis error after run the build_runner.
My codes was like:
```
enum PrinterType{
  PDF,
  ZPL,
  RAW
}
```
```
@Entity(tableName: 'Printer')
class PrinterEntity {
 final PrinterType? type;
}
```
Then the database.g.dart generated codes like:
```
(PrinterEntity item) => <String, Object?>{
                  'type': item.type == null ? null : item.type.index,
}
```
The error was: "**Error: Property 'index' cannot be accessed on 'PrinterType?' because it is potentially null**" and "**Context: 'type' refers to a property so it couldn't be promoted**".
I thought maybe replace `item.type == null ? null : item.type.index` with `item.type?.index` could fix.

**Comment by dkaera** [comment]
@xiaozhizhong good point,
thanks for the report!

**Comment by dkaera** [comment]
@xiaozhizhong fixed in #722

**Comment by xiaozhizhong** [comment]
> @xiaozhizhong fixed in #722

thanks! Will it be released to pub in the next few days?

**Comment by dkaera** [comment]
> > @xiaozhizhong fixed in #722
> 
> thanks! Will it be released to pub in the next few days?

yes, fix is in the main develop branch
should be published very soon.

**Comment by xiaozhizhong** [comment]
Okay，thank you agian. Let me close this issue.
